test/xmltest/__snapshots__/valid.test.js.snap

```diff
@@ -278,49 +274,50 @@ Object {
 exports[`xmltest/valid standalone should match 038.xml with snapshot 1`] = `
 Object {
   "actual": "<!-- comment -->
-<doc/>",
+<!DOCTYPE doc><doc/>",
   "expected": "<doc></doc>",
 }
 `;
 
 exports[`xmltest/valid standalone should match 039.xml with snapshot 1`] = `
 Object {
   "actual": "<?pi data?>
-<doc/>",
+<!DOCTYPE doc><doc/>",
   "expected": "<?pi data?><doc></doc>",
 }
 `;
 
 exports[`xmltest/valid standalone should match 040.xml with snapshot 1`] = `
 Object {
-  "actual": "<doc a1=\\"&quot;&lt;&amp;>'\\"/>",
+  "actual": "<!DOCTYPE doc><doc a1=\\"&quot;&lt;&amp;>'\\"/>",
   "expected": "<doc a1=\\"&quot;&lt;&amp;&gt;'\\"></doc>",
 }
 `;
 
 exports[`xmltest/valid standalone should match 041.xml with snapshot 1`] = `
 Object {
-  "actual": "<doc a1=\\"A\\"/>",
+  "actual": "<!DOCTYPE doc><doc a1=\\"A\\"/>",
   "expected": "<doc a1=\\"A\\"></doc>",
 }
 `;
 
 exports[`xmltest/valid standalone should match 042.xml with snapshot 1`] = `
 Object {
-  "actual": "<doc>A</doc>",
+  "actual": "<!DOCTYPE doc><doc>A</doc>",
+  "expected": "<doc>A</doc>",
```

Comment:
FYI: This is something I might consider a "failing test", since previously the lib was yielding the expected result (not added to the snapshot twice when `actual === expected`), and now there is additional stuff.
Of course we completely ignored/removed the doctype from the input, and now it's considered, so we might decie to live with it.

But I agree that this needs some checking.

Reply:
From all the information I could find from the perspective of specs, I couldn't find anything related to dropping the `DOCTYPE` under certain conditions:
https://w3c.github.io/DOM-Parsing/#dfn-xml-serializing-a-documenttype-node

I'm assuming the the "expected" value for some reason never contains the doctype, even if it' present in the input.
So from that perspective I'm assuming this isn't an issue.

Reply:
I was wondering about that too; i.e. why the tests are so defined. It seems the behaviour now (to output the doctype declaration even if empty) is in line with what both libxml2 and browsers do.

Reply:
The only clue I have is that the author of those test cases is also talking about "Canonical XML" and that this might be the reason why the doctype is not part of the expected output.

Reply:
The following is part of the `xmltest.zip` (file `canonxml.html`) bundled as part of the `xmltest` lib we use:

## Canonical XML
> This document defines a subset of XML called canonical XML. The intended use of canonical XML is in testing XML processors, as a representation of the result of parsing an XML document.
> 
> Every well-formed XML document has a unique structurally equivalent canonical XML document. Two structurally equivalent XML documents have a byte-for-byte identical canonical XML document. Canonicalizing an XML document requires only information that an XML processor is required to make available to an application.
> 
> A canonical XML document conforms to the following grammar:

```
CanonXML    ::= Pi* element Pi*
element     ::= Stag (Datachar | Pi | element)* Etag
Stag        ::= '<'  Name Atts '>'
Etag        ::= '</' Name '>'
Pi          ::= '<?' Name ' ' (((Char - S) Char*)? - (Char* '?>' Char*)) '?>'
Atts        ::= (' ' Name '=' '"' Datachar* '"')*
Datachar    ::= '&amp;' | '&lt;' | '&gt;' | '&quot;'
                 | '&#9;'| '&#10;'| '&#13;'
                 | (Char - ('&' | '<' | '>' | '"' | #x9 | #xA | #xD))
Name        ::= (see XML spec)
Char        ::= (see XML spec)
S           ::= (see XML spec)
```

> Attributes are in lexicographical order (in Unicode bit order).
> 
> A canonical XML document is encoded in UTF-8.
> 
> Ignorable white space is considered significant and is treated equivalently to data.

No doctype present, so we should drop it from the test output after serialization (instead of the test input), to get closer to the expected values.